frontend/public/components/utils/resource-log.jsx

```diff
@@ -164,316 +170,290 @@ LogControls.propTypes = {
+        k8sGet(ProjectModel, props.resource.metadata.namespace),
+      ];
+      Promise.all(promises)
+        .then(([podLogLinks_, project]) => {
+          // Project UID and namespace UID are the same value. Use the projects
+          // API since normal OpenShift users can list projects.
+          setPodLogLinks(podLogLinks_);
+          setNamespaceUID(project.metadata.uid);
+        })
+        .catch((e) => setError(e));
+    };
+
+    if (flags.CONSOLE_EXTERNAL_LOG_LINK && resource.kind === 'Pod') {
+      fetchLogLinks();
+    }
+    wsInit(props);
+    if (screenfull.enabled) {
+      screenfull.on('change', () => {
+        setIsFullscreen(screenfull.isFullscreen);
+      });
+      screenfull.on('error', () => {
+        setIsFullscreen(false);
+      });
     }
-    this.setState(newState);
-  }
 
-  // Destroy websocket
-  _wsDestroy() {
-    this.ws && this.ws.destroy();
-  }
+    return () => {
+      wsDestroy();
+      if (screenfull.enabled) {
+        screenfull.off('change');
+        screenfull.off('error');
+      }
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  // Initialize websocket connection and wire up handlers
-  _wsInit({ resource, containerName, bufferSize }) {
-    if (
-      [LOG_SOURCE_RUNNING, LOG_SOURCE_TERMINATED, LOG_SOURCE_RESTARTING].includes(
-        this.state.resourceStatus,
-      )
-    ) {
-      const urlOpts = {
-        ns: resource.metadata.namespace,
-        name: resource.metadata.name,
-        path: 'log',
-        queryParams: {
-          container: containerName || '',
-          follow: 'true',
-          tailLines: bufferSize,
-        },
-      };
-      const watchURL = resourceURL(modelFor(resource.kind), urlOpts);
-      const wsOpts = {
-        host: 'auto',
-        path: watchURL,
-        subprotocols: ['base64.binary.k8s.io'],
-      };
+  React.useEffect(() => {
+    if (props.resourceStatus !== prevResourceStatus) {
+      const newState = {};
+      newState.resourceStatus = props.resourceStatus;
+      // Container changed from non-running to running state, so currently displayed logs are stale
+      if (
+        prevResourceStatus === LOG_SOURCE_RESTARTING &&
+        resourceStatus !== LOG_SOURCE_RESTARTING
+      ) {
+        newState.stale = true;
+      }
+      setPrevResourceStatus(resourceStatus);
+      setResourceStatus(newState.resourceStatus);
+      newState.stale && setStale(newState.stale);
+    }
+  }, [props.resourceStatus, prevResourceStatus, resourceStatus]);
 
-      this.ws = new WSFactory(watchURL, wsOpts)
-        .onclose(this._onClose)
-        .onerror(this._onError)
-        .onmessage(this._onMessage)
-        .onopen(this._onOpen);
+  React.useEffect(() => {
+    const resourceStarted =
+      prevResourceStatus === LOG_SOURCE_WAITING && resourceStatus !== LOG_SOURCE_WAITING;
+    if (resourceStarted) {
+      restartStream();
     }
-  }
+  }, [prevResourceStatus, resourceStatus, restartStream]);
```

Comment:
```suggestion
```

Reply:
Did you mean to suggest something here? It looks identical.